File: src/homework/Nov26/miniProject/ProductManager.java
package homework.Nov26.miniProject;

import homework.Nov26.miniProject.Interface.IManager;

import java.util.ArrayList;
import java.util.List;

public class ProductManager implements IManager {
	
	private final List<Product> products;
	
	public ProductManager() {
		products = new ArrayList<>();
	}
	
	public void addProduct(Product product) {
		products.add(product);
	}
	
	public int getSize() {
		return products.size();
	}
	
	public Product getProduct(int id) {
		for(Product p: products) {
			if(p.getId() == id) {
				return p;
			}
		}
		
		return null; 
	}
	
	public void showProducts() {
		if(products.isEmpty()) {
			System.out.println("Empty List");
		} else {
			System.out.println("=========== Product List ===========");
		}
		for(Product product: products) {
			System.out.println(product);
		}
		System.out.println("");
	}

	@Override
	public void deleteProduct(Product product) {
		products.remove(product);
	}

	@Override
	public boolean isContain(Product product) {
		return products.contains(product);
	}
	
	public Product getProductById(int id) {
		for(Product product: products) {
			if(product.getId() == id) {
				return product;
			}
		}
		return null;
	}
}

File: src/lab/miniProject_TS/src/productManager.ts
import { IManager } from './interface'
import Product from './product'

class ProductManager implements IManager {
	
	private products:Array<Product>
	
	constructor() {
		this.products = new Array<Product>();
	}

    public deleteProduct(product: Product): void {
        this.products = this.products.filter(p => p.getId() !== product.getId())
	}
	
    public showProducts(): void {
		let message: string = this.products.length === 0 ? 'Empty List' : '=========== Product List ==========='
		console.log(message)
		this.products.map(p => console.log(p.toString()))
	}
	
    public isContain(product: Product | null): boolean {
		if(product === null) return false
		return this.products.filter(p => p.getId() === product.getId()).length > 0
	}
	
	public addProduct(product: Product): void {
		this.products.push(product)
	}
	
	public getSize(): number {
		return this.products.length
	}
	
	public getProduct(id: number): Product {
        let product = this.products.filter(p => p.getId() === id)
        return product[0]
	}
	
}

export default ProductManager
